src/components/Reports/GoalCalculator/Shared/GoalCalculatorSection.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import InfoIcon from '@mui/icons-material/Info';
+import { IconButton, Typography, styled } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { useGoalCalculator } from './GoalCalculatorContext';
+
+const SectionContainer = styled('div')(({ theme }) => ({
+  paddingInline: theme.spacing(4),
+}));
+
+interface GoalCalculatorSectionProps {
+  title: string;
+  rightPanelContent?: JSX.Element;
+  children: React.ReactNode;
+}
+
+export const GoalCalculatorSection: React.FC<GoalCalculatorSectionProps> = ({
+  title,
+  rightPanelContent,
+  children,
+}) => {
+  const { setRightPanelContent } = useGoalCalculator();
+  const { t } = useTranslation();
+
+  return (
```

Comment:
[Mentoring] Well-designed optional prop pattern. The `hasInfoPanel` boolean with conditional info icon rendering provides clean API design. This demonstrates good component composition - the section can work independently or integrate with info panels without tight coupling.